v1/topdown/eval.go

```diff
@@ -2019,7 +2019,26 @@ func (e evalFunc) eval(iter unifyIterator) error {
 		return e.e.saveCall(argCount, e.terms, iter)
 	}
 
-	if e.e.partial() && (e.e.inliningControl.shallow || e.e.inliningControl.Disabled(e.ref, false)) {
+	var generateSupport bool
+
+	if rule := e.ir.Default; rule != nil {
+		if len(rule.Head.Args) == len(e.terms)-2 {
+			// If the output term is not constant OR it's equal to the default value, then
+			// a support rule must be produced as the default value _may_ be required. On
+			// the other hand, if the output term is constant (i.e., it does not require
+			// evaluation) and it differs from the default value then the default value is
+			// _not_ required, so partially evaluate the rule normally.
+			resultVar := e.terms[len(e.terms)-1]
+			rterm := e.e.bindings.Plug(resultVar)
```

Comment:
Should this not still be wrapped in a `e.e.partial()` check? 
If this applied to both PE and normal eval, does the comment about creating a support rule apply to both?

Missing `e.e.partial()` checks previously meant we did some very expensive things in eval which only applied to PE, and that's why I'm asking 🙂

Reply:
We could optimize here by wrapping everything in a `e.e.partial()` check, correct 👍.

But as it stands, this entire block is superfluous because of misaligned preconditions. I want do decide if this should be kept in some form before optimizing/fixing minor issues.